internal/retry/retry_test.go

```diff
@@ -540,20 +593,127 @@ func TestMiddleware(t *testing.T) {
 				},
 			},
 		},
+		{
+			msg: "multiple retry policies",
+			policyProvider: newProcedurePolicyProvider().registerDefault(
+				NewPolicy(
+					Retries(2),
+					MaxRequestTimeout(time.Millisecond*20),
+					BackoffStrategy(newFixedBackoff(time.Millisecond*25)),
+				),
+			).registerService(
+				"serviceRoute",
+				NewPolicy(
+					Retries(1),
+					MaxRequestTimeout(time.Millisecond*50),
+					BackoffStrategy(newFixedBackoff(time.Millisecond*50)),
+				),
+			).registerServiceProcedure(
+				"serviceRoute",
+				"procedureRoute",
+				NewPolicy(
+					Retries(0),
+					MaxRequestTimeout(time.Millisecond*100),
+				),
+			).registerServiceProcedure(
+				"serviceRoute",
+				"fakeProcedureRoute",
+				NewPolicy(
+					Retries(100),
+					MaxRequestTimeout(time.Millisecond*10000),
+				),
+			),
+			actions: []MiddlewareAction{
+				ConcurrentAction{
+					Actions: []MiddlewareAction{
+						RequestAction{
+							request: &transport.Request{
+								Service:   "nonServiceRoute",
+								Procedure: "nonProcedureRoute",
+								Body:      bytes.NewBufferString("body1"),
+							},
+							reqTimeout: time.Millisecond * 200,
+							events: []*OutboundEvent{
+								{
+									WantTimeout:    time.Millisecond * 20,
+									WantService:    "nonServiceRoute",
+									WantProcedure:  "nonProcedureRoute",
+									WantBody:       "body1",
+									WaitForTimeout: true,
+									GiveError:      errors.ClientTimeoutError("serv", "proc", time.Millisecond*20),
+								},
+								{
+									WantTimeout:    time.Millisecond * 20,
+									WantService:    "nonServiceRoute",
+									WantProcedure:  "nonProcedureRoute",
+									WantBody:       "body1",
+									WaitForTimeout: true,
+									GiveError:      errors.ClientTimeoutError("serv", "proc", time.Millisecond*20),
+								},
+								{
+									WantTimeout:   time.Millisecond * 20,
+									WantService:   "nonServiceRoute",
+									WantProcedure: "nonProcedureRoute",
+									WantBody:      "body1",
+									GiveError:     errors.ClientTimeoutError("serv", "proc", time.Millisecond*20),
+								},
+							},
+							wantError: errors.ClientTimeoutError("serv", "proc", time.Millisecond*20).Error(),
+						},
+						RequestAction{
```

Comment:
This one test case is exercising like 5 different things. Can we pull this into a separate test function where you're reusing the policy provider?